### PR TITLE
[CI] Add the `-no_warn_duplicate_libraries` ldflag when running inv test on `XCode >15`

### PR DESCRIPTION
**Pull request by amenasria**
### What does this PR do?

This PR adds the `-no_warn_duplicate_libraries` flag when running inv test on XCode >15.

### Motivation

If you currently run the inv test invoke task on a MacOS machine with XCode 15 you’ll encounter some warning messages. Warnings should in theory be fine but we’re using gotestsum which is handling warning as errors, breaking its run [1]. Examples:
```
> inv -e test --rerun-fails=2 --coverage
=== FAIL: comp/trace/config TestMockConfig (2.29s)
    writer.go:40: [Fx] ...
    config_test.go:2526: 
                Error Trace:    /Users/alexandre.menasria/datadog-agent/comp/trace/config/config_test.go:2526
                Error:          Should not be: "foo"
                Test:           TestMockConfig

=== Errors
ld: warning: ignoring duplicate libraries: '-ldatadog-agent-rtloader', '-ldl'
...
ld: warning: ignoring duplicate libraries: '-ldatadog-agent-rtloader', '-ldl'
```

```
> inv -e test --rerun-fails=2 --coverage  
Using default modules and targets
--- Setting rtloader paths to lib:/Users/alexandre.menasria/datadog-agent/dev/lib | header:/Users/alexandre.menasria/datadog-agent/dev/include | common headers:
--- Flavor base: unit tests
-----  Module '/Users/alexandre.menasria/datadog-agent'
cd /Users/alexandre.menasria/datadog-agent && gotestsum  --jsonfile "module_test_output.json"  --format pkgname --rerun-fails=2 --packages="./pkg/... ./cmd/... ./comp/..." --raw-command /Users/alexandre.menasria/datadog-agent/test_with_coverage.sh -- -mod=mod -tags "kubelet ec2 oracle apm clusterchecks kubeapiserver consul jmx process gce fargateprocess python test orchestrator etcd otlp zlib zk" -gcflags="" -ldflags="-X github.com/DataDog/datadog-agent/pkg/version.Commit=165f0d4a19 -X github.com/DataDog/datadog-agent/pkg/version.AgentVersion=7.53.0-devel+git.191.165f0d4 -X github.com/DataDog/datadog-agent/pkg/serializer.AgentPayloadVersion=v5.0.105 -X github.com/DataDog/datadog-agent/pkg/config/setup.ForceDefaultPython=true -X github.com/DataDog/datadog-agent/pkg/config/setup.DefaultPython=3 -r /Users/alexandre.menasria/datadog-agent/dev/lib '-extldflags=-Wl,-bind_at_load ' "   -vet=off  -timeout 180s -short -covermode=count  
✓  pkg/aggregator (6.615s) (coverage: 67.5% of statements)
✓  pkg/aggregator/internal/tags (683ms) (coverage: 91.4% of statements)
✓  pkg/aggregator/internal/util (328ms) (coverage: 100.0% of statements)
✓  pkg/aggregator/mocksender (744ms) (coverage: 61.7% of statements)
∅  pkg/aggregator/sender
✓  pkg/autodiscovery (1.051s) (coverage: 70.7% of statements)
✓  pkg/autodiscovery/common/types (312ms) (coverage: 46.3% of statements)
✓  pkg/autodiscovery/common/utils (686ms) (coverage: 84.1% of statements)
✓  pkg/autodiscovery/configresolver (776ms) (coverage: 87.5% of statements)
✓  pkg/autodiscovery/integration (339ms) (coverage: 64.0% of statements)
✓  pkg/autodiscovery/listeners (3.006s) (coverage: 41.7% of statements)
∅  pkg/autodiscovery/providers/names
✓  pkg/autodiscovery/scheduler (680ms) (coverage: 80.0% of statements)
∅  pkg/autodiscovery/telemetry
∅  pkg/cli/standalone
# github.com/DataDog/datadog-agent/pkg/cli/subcommands/check.test
ld: warning: ignoring duplicate libraries: '-ldatadog-agent-rtloader', '-ldl'
✓  pkg/cli/subcommands/check (1.03s) (coverage: 12.4% of statements)
✓  pkg/autodiscovery/providers (6.094s) (coverage: 60.8% of statements)
# github.com/DataDog/datadog-agent/pkg/cli/subcommands/clusterchecks.test
ld: warning: ignoring duplicate libraries: '-ldatadog-agent-rtloader', '-ldl'
✓  pkg/cli/subcommands/clusterchecks (882ms) (coverage: 26.2% of statements)
✓  pkg/cli/subcommands/config (1.238s) (coverage: 20.3% of statements)
# github.com/DataDog/datadog-agent/pkg/cli/subcommands/dcaconfigcheck.test
ld: warning: ignoring duplicate libraries: '-ldatadog-agent-rtloader', '-ldl'
# github.com/DataDog/datadog-agent/pkg/cli/subcommands/dcaflare.test
ld: warning: ignoring duplicate libraries: '-ldatadog-agent-rtloader', '-ldl'
✓  pkg/cli/subcommands/dcaconfigcheck (897ms) (coverage: 85.7% of statements)
```

This most likely happens because the `-ldl` and `-ldatadig-agent-rtloader` flags are added in several places (e.g in [pkg/collector/python/aggregator.go](https://github.com/DataDog/datadog-agent/blob/de157bfec3ac33c10e0812889162619a85368875/pkg/collector/python/aggregator.go/#L20-L25) and in [pkg/collector/python/datadog_agent.go](https://github.com/DataDog/datadog-agent/blob/de157bfec3ac33c10e0812889162619a85368875/pkg/collector/python/datadog_agent.go/#L30-L36)). 

[From this article it seems that this warning was not silent anymore since Xcode 15](https://indiestack.com/2023/10/xcode-15-duplicate-library-linker-warnings/). 

[1] Runs are broken like this:
- If no test fails but the warning appears `gotestsum` will return an exit code != 0.
- If a test fails and the warning appears `gotestsum` won’t rerun the test (because it doesn’t rerun on failure outside of the test).

### CODE REVIEW

File: tasks/libs/common/utils.py [hunk 1]
```diff
@@ -228,10 +228,25 @@ def get_build_flags(
     elif os.environ.get("NO_GO_OPT"):
         gcflags = "-N -l"
 
-    # On macOS work around https://github.com/golang/go/issues/38824
-    # as done in https://go-review.googlesource.com/c/go/+/372798
     if sys.platform == "darwin":
-        extldflags += "-Wl,-bind_at_load "
+        # On macOS work around https://github.com/golang/go/issues/38824
+        # as done in https://go-review.googlesource.com/c/go/+/372798
+        extldflags += "-Wl,-bind_at_load,-no_warn_duplicate_libraries"
```
Comment (spencergilbert) [inline tasks/libs/common/utils.py:234]: This looks like it's adding it by default and then adding it a second time in the conditional?

Comment (amenasria) [reply tasks/libs/common/utils.py:234]: Oups forgot to remove this, thanks ! Fixing it !

**Comment by pgimalac** [comment]
Tested this locally and it worked properly, no more `ld` warnings 👍

**Review by spencergilbert — APPROVED** [review]
LGTM

**Pull request merged**